numpy/lib/function_base.py

```diff
@@ -108,20 +115,35 @@ def rice(x):
         The number of bins is proportional to the cube root of data size (asymptotically optimal)
         Depends only on size of the data
         """
-        return np.ceil(2 * x.size ** (1.0 / 3))
+        return np.ceil(2 * np.cbrt(x.size))
 
     def scott(x):
         """
         Scott Estimator
         The binwidth is proportional to the standard deviation of the data and
         inversely proportional to the cube root of data size (asymptotically optimal)
-
         """
-        h = 3.5 * x.std() * x.size ** (-1.0 / 3)
+        h = 3.5 * x.std() / np.cbrt(x.size)
         if h > 0:
             return np.ceil(x.ptp() / h)
         return 1
 
+    def doane(x):
+        """
+        Doane's Estimator
+        Improved version of Sturges' formula which works better for non-normal
+        data.
+        See http://stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
```

Comment:
While I would normally agree with you, the author of the better researched Stack Exchange answer is also the author of the relevant section on Wikipedia. I referenced the Stack Exchange article because it is not only the original source, but also contains a lot more information.